app.py

```diff
@@ -41,22 +41,100 @@ def teams_create():
-# - For teams, items should be ranked higher if any of the words in the search query are in the name or description
-# - Not all properties are treated equally: if there is a match in the name, rank the item higher than other properties
+# (done) Searches should be case insensitive
+# (done) You should not use JavaScript or AJAX in search.html
+# (done) For pokemon, items should be ranked higher if any of the words in the search query are in the name, description, or types
+# (done) For teams, items should be ranked higher if any of the words in the search query are in the name or description
+# (done) Not all properties are treated equally: if there is a match in the name, rank the item higher than other properties
 
 # Extra requirements when you're done:
-# - Teams should rank higher if their member pokemon also match
-# - Highlight the matching word in the name or description in the search results
-# - Implement a filter for search queries (e.g. searching "p type:fire" will only
+# (done) Teams should rank higher if their member pokemon also match
+# (done) Highlight the matching word in the name or description in the search results
+# (part done) Implement a filter for search queries (e.g. searching "p type:fire" will only
 #   search for pokemon that contain the letter p that are fire type pokemon)
 # - Add pagination to your search results page so only 20 pokemon show up at a time,
 #   but you can navigate to another page
+
+# takes a string like "p type:electric" and splits it into p, type, and electric
+def search_parser(search_string):
+        filters = re.match(r'^(?P<str>.*)\s(?P<mod>.*)[:](?P<fil>.*)(\s|$)', search_string)
+        if filters != None:
+            return filters
+        else:
+            return False
+
+# generator function that returns at the first match or yields until it's called again
+def search_db(search_string, search_object):
+    if search_string in search_object['name'].lower():
+        return search_object
+    else:
+        yield
+    if search_string in search_object['description'].lower():
+        return search_object
+    else:
+        yield
+    if 'types' in search_object.keys():
+        for t in search_object['types']:
+            if search_string == t.lower():
+                return search_object
+        return False
+    else:
+        for p in search_object['members']:
+            if search_string == pokemon[p['pokemon_id']-1]['name'].lower():
+                return search_object
+        return False
+
+def highlighter(results, search_string, keyArray):
+    for i in range(len(results)):
+        for j in keyArray:
+            caps_string = search_string.capitalize()
+            results[i][j] = results[i][j].replace(search_string,f'<span class="highlight">{search_string}</span>')
+            results[i][j] = results[i][j].replace(caps_string,f'<span class="highlight">{caps_string}</span>')
```

Comment:
It looks like this only accounts for lowercase and capitalized versions of your search string. What about mixed caps? Could you use a regex here to make it more resilient?